jax/_src/api.py

```diff
@@ -177,10 +177,52 @@ def wrapped(*args, **kwargs):
     return f(*args, **kwargs)
   return cast(F, wrapped)
 
+class NoFunctionType:
```

Comment:
nit: Should this be marked as private? Also the instance `NoFunction`?

Reply:
It's in `jax/_src`, so it's already private.